picard/oauth.py

```diff
@@ -172,28 +177,37 @@ def set_access_token(self, access_token, expires_in):
         self.access_token = access_token
         self.access_token_expires = int(time.time() + expires_in - 60)
 
-    def refresh_access_token(self, callback):
-        log.debug("OAuth: refreshing access_token with a refresh_token %s", self.refresh_token)
-        path = "/oauth2/token"
+    @staticmethod
+    def _query_data(params):
         url = QUrl()
         url_query = QUrlQuery()
-        url_query.addQueryItem("grant_type", "refresh_token")
-        url_query.addQueryItem("refresh_token", self.refresh_token)
-        url_query.addQueryItem("client_id", MUSICBRAINZ_OAUTH_CLIENT_ID)
-        url_query.addQueryItem("client_secret", MUSICBRAINZ_OAUTH_CLIENT_SECRET)
+        for key, value in params.items():
+            url_query.addQueryItem(key, value)
         url.setQuery(url_query.query(QUrl.FullyEncoded))
-        data = url.query()
-        self.webservice.post(self.host, self.port, path, data,
-                             partial(self.on_refresh_access_token_finished, callback),
-                             mblogin=True, priority=True, important=True,
-                             request_mimetype="application/x-www-form-urlencoded")
+        return url.query()
+
+    def refresh_access_token(self, callback):
+        log.debug("OAuth: refreshing access_token with a refresh_token %s", self.refresh_token)
+        path = "/oauth2/token"
+        params = {
+            "grant_type": "refresh_token",
+            "refresh_token": self.refresh_token,
+            "client_id": MUSICBRAINZ_OAUTH_CLIENT_ID,
+            "client_secret": MUSICBRAINZ_OAUTH_CLIENT_SECRET,
+        }
+        self.webservice.post(
+            self.host, self.port, path, self._query_data(params),
```

Comment:
Instead of `self._query_data` we could also use `urllib.parse.urlencode` and get rid of the custom Qt based implementation.

Reply:
Yup, agreed.

See d1de2a7208 and 8b787482f